02-image-classification.Rmd

```diff
@@ -11,165 +11,190 @@ cat(author)
 
-## Task 5: Creating a Map {.unnumbered}
+- Urban
+- Water
+- Forest
+- Agriculture
 
-**Making your map**
+**Step 1:** First we will create a sample of pixels to train the classification on. In the Geometry Imports menu, click **+new layer**. Add a new layer for each of the four land cover classes, rename each layer and assign it a color using the gear icon. 
 
-Now is the time for your cartographic skills to shine. To design a map of the highest quality you will need to draw on all of you GIS skills. Ensure you include all the standard components of a map along with your artistic flair.
+Change Import as to **Feature Collection.**  
+Add a new property called **landcover** and fill in a **numeric value** for each land cover type. For example, Water = 0, Forest = 1, Urban = 2 etc...
 
-You should design a map that shows your land cover classification overtop of the provided Landsat Imagery. Pick appropriate colours and transparency for each layer.
+``` {r, echo=FALSE, out.width = "75%", fig.align = 'center'}
+knitr::include_graphics('images/02/training.jpg')
+```
 
-Use inset maps to show the location of the study area within western Canada and to show a zoomed in subset of your classification map in greater detail. 
+Select a land cover class in the Geometry Imports menu and use the point and polygon tools to select pixels in the image representing that class. Toggle the NDVI and true color composite on/off to aid your land cover interpretation. The example below gives an idea of how many pixels you want to sample per class. 
 
-The physical requirements are as follows: 
+``` {r, echo=FALSE, fig.align = 'center'}
+knitr::include_graphics('images/02/sample.jpg')
+```
 
-- Map should be 11”x17” either as a landscape or portrait layout 
-- You should export the map as a pdf 
-- Your map should incorporate the suite of standard map elements: Title, scale bar, compass, legend and appropriate inset maps. 
+**Step 2:**Once you have selected training areas for each land cover class use the following code block to merge the Feature Collections and select a sample of pixels to train the classification model. The training pixels will be stored as a new variable called `training`.
+
+    var classNames = Water.merge(Urban).merge(Agriculture).merge(Forest);
+    print(classNames);
+    
+    var bands = ['SR_B2', 'SR_B3', 'SR_B4', 'SR_B5', 'SR_B6', 'SR_B7', 'NDVI'];
+    var training = image.select(bands).sampleRegions({
+      collection: classNames,
+      properties: ['landcover'],
+      scale: 30
+    });
+
+##### Q5: Explain how a supervised and unsupervised classification are different? {.unnumbered}
+
+**Step 3:** Next we will use the **Random Forests** algorithm to train our classifier and predict land cover for all the pixels in our image. First we make a new variable for the model trained on our sample of pixels, then we apply the model to image and save it as a new image called `classified`. 
```

Comment:
Random forest is too advanced of an algorithm and out of scope for NRES 341. We should stick to something basic like maximum likelihood.